lib/storekit/client.rb

```diff
@@ -28,12 +28,13 @@ def verify!(base64_payload)
       resp = @http.request(req)
 
       decoded = JSON.parse resp.body
+      @last_response = ReceiptResponse.new(decoded)
```

Comment:
> Why should, it is an action - method with the exclamation mark `!`, so it can modify the host object.

The convention of adding `!` to a method name to indicate the method's behaviour is open-ended. Yes, it is true that sometimes it is used to indicate that the method modifies the receiver, like `String#strip!`, but in some other cases it indicates that the method can raise an error, like `ActiveRecord::FinderMethods#find_by!`.

I should have stated that adding this attribute breaks the client object's thread safety.